types/src/dkg.rs

```diff
@@ -54,3 +69,63 @@ impl OnChainConfig for DKGState {
     const MODULE_IDENTIFIER: &'static str = "dkg";
     const TYPE_IDENTIFIER: &'static str = "DKGState";
 }
+
+pub trait DKGTrait {
+    type PrivateParams;
+    type PublicParams: Send + Sync;
+    type Transcript: Clone + Default + Send + Sync + for<'a> Deserialize<'a>;
+
+    fn generate_transcript<R: CryptoRng>(
+        rng: &mut R,
+        sk: &Self::PrivateParams,
+        params: &Self::PublicParams,
+    ) -> Self::Transcript;
+
+    fn verify_transcript(params: &Self::PublicParams, trx: &Self::Transcript) -> Result<()>;
+
+    fn aggregate_transcripts(
+        params: &Self::PublicParams,
+        base: &mut Self::Transcript,
+        extra: &Self::Transcript,
+    );
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct DummyDKGTranscript {
```

Comment:
Make it test only

Reply:
fixed

Reply:
not yet, before real transcript is available.